packages/api-tests/test/window.core.spec.js

```diff
@@ -331,6 +358,200 @@ describe('WindowCore API', function(done) {
+      const windowTitle = 'windownameremovelistener';
+      const windowOptions = getWindowOptions({
+        name: windowTitle
+      });
+
+      const addWindowListener = (event) => {
+        /* eslint-disable no-undef */
+        const script = (event, callback) => {
+          // We need to save the function, as we need to pass the same function object to removeListener
+          window.customListener = () => console.log(event);
+          var currentWin = ssf.Window.getCurrentWindow();
+          currentWin.addListener(event, window.customListener);
+          callback();
+        };
+        /* eslint-enable no-undef */
+        return executeAsyncJavascript(app.client, script, event);
+      };
+
+      const removeWindowListener = (event) => {
+        /* eslint-disable no-undef */
+        const script = (event, callback) => {
+          var currentWin = ssf.Window.getCurrentWindow();
+          currentWin.removeListener(event, window.customListener);
+          callback();
+        };
+        /* eslint-enable no-undef */
+        return executeAsyncJavascript(app.client, script, event);
+      };
+
+      const steps = [
+        ...setupWindowSteps(windowOptions),
+        () => addWindowListener('blur'),
+        () => removeWindowListener('blur'),
+        () => callWindowMethod('listenerCount', 'blur'),
+        (result) => assert.equal(result.value, 0)
+      ];
+
+      return chainPromises(steps);
+    });
+
+    it.only('Should remove all listeners from the window #ssf.Window.removeAllListeners #ssf.WindowCore.removeAllListeners', function() {
+      const windowTitle = 'windownameremovealllisteners';
+      const windowOptions = getWindowOptions({
+        name: windowTitle
+      });
+
+      const addWindowListener = (event, data) => {
+        /* eslint-disable no-undef */
+        const script = (event, data, callback) => {
+          // We need to save the function, as we need to pass the same function object to removeListener
+          var currentWin = ssf.Window.getCurrentWindow();
+          currentWin.addListener(event, () => console.log(event + data));
+          callback();
+        };
+        /* eslint-enable no-undef */
+        return executeAsyncJavascript(app.client, script, event, data);
+      };
+
+      const removeWindowListeners = (event) => {
+        /* eslint-disable no-undef */
+        const script = (event, callback) => {
+          var currentWin = ssf.Window.getCurrentWindow();
+          currentWin.removeAllListeners(event);
+          callback();
+        };
+        /* eslint-enable no-undef */
+        return executeAsyncJavascript(app.client, script, event);
+      };
+
+      const steps = [
+        ...setupWindowSteps(windowOptions),
+        () => addWindowListener('blur', 'one'),
+        () => addWindowListener('blur', 'two'),
+        () => removeWindowListeners('blur'),
```

Comment:
Maybe we could have `() => addWindowListener('focus', 'one')` here, and verify that the listener for `focus` **doesn't** get removed?